### ISSUE TITLE
Add VICReg Model

### ISSUE THREAD

**Issue by felipevillaarenas** [Lightning-AI/lightning-bolts#889]
## 🚀 Feature

Pytorch lightning implementation of the **VICReg model** following the style of other SSL models already available in PL Bolts.

### Motivation

[VICReg](https://arxiv.org/pdf/2105.04906.pdf)  is a **SOTA** model that is frequently used in Self Supervised Learning benchmarks but is not available in PL Bolts.

This method avoids the collapse problem with two regularizations terms applied to both embeddings separately: (1) a term that maintains the variance of each embedding dimension above a threshold, (2) a term that decorrelates each pair of variables. Unlike most other approaches to the same problem, VICReg does not require techniques such as: weight sharing between the branches, batch normalization, feature-wise normalization, output quantization, stop gradient, memory banks, etc., and achieves results on par with the state of the art on several downstream tasks. 

### Pitch

I already ported my code of VICReg from PyTorch to PyTorch Lightning. Also, I wrote the module trying to keep the style used in PL bolts for other SOTA Self-Supervised models (e.g. simsiam, simclr). 

Then I would like to add this module to the PyTorch Lightning Bolts repo. You can find the code on my [repo](https://github.com/felipevillaarenas/vicreg).

> Additionally, similar to other SSL models in PL Bolts the module supports the following datasets: cifar10, stl10 and imagenet.

### Additional context
I have pre-train the model for CIFAR10([here WandB eval metrics for CIFAR10](https://wandb.ai/felipevilla/VICReg-CIFAR10/reports/VICReg-Pre-Training-CIFAR10-And-Online-Eval--VmlldzoyNjc3MjUw?accessToken=dax8gmjqky9xl8peo8s2gu266sxq33rgdzuqshpjl5mc9sqf80eplnyqnjue2zn5)) but pertaining of stl10 and imagenet are still pending

**Comment by otaj** [comment]
Hi @felipevillaarenas, thank you for your suggestion! Currently, we're not accepting new features as we're going through a major revision (see #839). However, we'd be happy to accept it once the revision is done. In the meantime, you can contribute through revising the current code! :rocket: :zap:

**Comment by felipevillaarenas** [comment]
Hi @otaj, thanks for your reply.
